src/lib/mdns/minimal/QueryReplyFilter.h

```diff
@@ -59,9 +57,41 @@ class QueryReplyFilter : public ReplyFilter
         return *this;
     }
 
+    QueryReplyFilter & SetSendingAdditionalItems(bool additional)
+    {
+        mSendingAdditionalItems = additional;
+        return *this;
+    }
+
 private:
+    bool AcceptableQueryType(QType qType)
+    {
+        if (mSendingAdditionalItems && mQueryData.IsBootAdvertising())
```

Comment:
Is this check needed (isn't the 2nd condition true already?)

Reply:
IsBootAdvertising is only set during initial boot, not during normal query replies, where as 'sending additional items'  is set for any reply processing. I believe they are both needed.